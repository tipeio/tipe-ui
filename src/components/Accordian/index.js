export { default } from './Accordian.vue'
