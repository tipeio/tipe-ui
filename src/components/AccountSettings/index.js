export { default } from './AccountSettings.vue'
