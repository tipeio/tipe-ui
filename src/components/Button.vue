<template>
  <button
    :class="classObject"
    @click="onClick"
  >
    <slot />
  </button>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeButton',
  props: {
    size: vueTypes.oneOf(['small', 'medium', 'large']).def('medium'),
    outline: vueTypes.bool.def(false),
    color: vueTypes.oneOf(['purple', 'danger', 'dark-purple']).def('purple')
  },
  computed: {
    classObject: function() {
      const { color, size, outline } = this
      return {
        [size]: true,
        [color]: true,
        outline
      }
    }
  },
  methods: {
    onClick() {
      this.$emit('click')
    }
  }
}
</script>

<style lang="postcss" scoped>
@define-mixin outline $color {
  border: 0.5px solid $color;
  color: $color;
  background-color: #fff;
}

button {
  padding: 10px 20px;
  margin: 10px;
  border-radius: 5px;
  border: none;
  color: #fff;

  &.purple {
    background-image: var(--purple-gradient);
    &.outline {
      background-image: none;
      @mixin outline var(--purple);
    }
  }

  &.dark-purple {
    background-color: var(--purple-dark);
    &.outline {
      @mixin outline var(--purple-dark);
    }
  }

  &.danger {
    background-color: var(--danger);
    &.outline {
      @mixin outline var(--danger);
    }
  }

  &.small {
    font-size: 0.75rem;
  }
  &.medium {
    font-size: 1.25rem;
  }

  &.large {
    font-size: 1.5rem;
  }
}
</style>
