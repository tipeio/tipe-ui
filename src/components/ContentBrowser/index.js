export { default } from './ContentBrowser.vue'
