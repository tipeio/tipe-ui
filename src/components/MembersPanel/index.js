export TableRow from './TableRow'
export TipeTable from './Table'
export { default } from './MembersPanel'
