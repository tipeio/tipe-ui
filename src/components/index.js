export const components = {
  Accordian: require('./Accordian'),
  AvatarUpload: require('./AvatarUpload'),
  Breadcrumbs: require('./Breadcrumbs'),
  Button: require('./Button'),
  Card: require('./Card'),
  Checkbox: require('./Checkbox'),
  Dialog: require('./Dialog'),
  DocumenEditor: require('./DocumentEditor'),
  DropDown: require('./DropDown'),
  Field: require('./Field'),
  Finder: require('./Finder'),
  Icon: require('./Icon'),
  IconBadge: require('./IconBadge'),
  Image: require('./Image'),
  Markdown: require('./Markdown'),
  Message: require('./Modal'),
  NumberInput: require('./NumberInput'),
  ProfileImage: require('./ProfileImage'),
  Progress: require('./Progress'),
  Radio: require('./Radio'),
  Scrolltable: require('./Scrollable'),
  Search: require('./Search'),
  Select: require('./Select'),
  SettingsMenu: require('./SettingsMenu'),
  Sidebar: require('./Sidebar'),
  Switch: require('./Switch'),
  Tabs: require('./Tabs'),
  TabsPanel: require('./TabsPanel'),
  TextArea: require('./Textarea'),
  TextInput: require('./TextInput')
}
