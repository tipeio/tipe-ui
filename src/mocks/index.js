export createManyMocks from './createManyMocks'
export user from './User'
export link from './Link'
export navLink from './NavLink'
export role from './Role'
export file from './File'
