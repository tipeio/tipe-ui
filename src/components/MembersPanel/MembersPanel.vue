<template>
  <layout-card>
    <div class="grid">
      <div class="flex-row">
        <div class="flex-column">
          <p class="text">Members</p>
          <p class="small-text">Description text goes here</p>
        </div>
        <tipe-button icon-before="plus">Add Member</tipe-button>
      </div>
      <tipe-table :members="members" />
    </div>
  </layout-card>
</template>

<script>
import { TipeTable } from '.'
import { createManyMocks, user } from '../../mocks'
import LayoutCard from '../LayoutCard'
import TipeButton from '../Button'

export default {
  name: 'MembersPanel',
  components: { LayoutCard, TipeTable, TipeButton },
  data() {
    return {
      members: createManyMocks(user, 12)
    }
  }
}
</script>


<style lang='postcss' scoped>
.grid {
  display: grid;
  grid-template-rows: 4rem auto;
  width: 100%;
}

.flex-row {
  display: flex;
  max-width: 100%;
  justify-content: space-between;
  align-items: center;
  padding: 1.125rem;

  & .text {
    color: var(--darkPurple);
    font-size: 0.875rem;
    margin: 0;
    margin-bottom: 3px;
  }

  & .small-text {
    color: var(--midPurple);
    font-size: 0.75rem;
    margin: 0;
  }
}
.flex-column {
  display: flex;
  flex-direction: column;
}
</style>
