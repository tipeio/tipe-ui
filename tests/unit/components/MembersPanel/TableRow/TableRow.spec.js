import { shallowMount } from '@vue/test-utils'
import { TableRow } from '@/components/MembersPanel'
import { createManyMocks, user } from '../../../../../src/mocks'

const member = () => createManyMocks(user, 1)

describe('TableRow.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(TableRow, {
      propsData: {
        member: member()[0]
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
