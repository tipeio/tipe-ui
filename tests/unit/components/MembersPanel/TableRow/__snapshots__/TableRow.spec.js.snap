// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow.vue renders 1`] = `
<div class="padding">
  <tr class="member-row">
    <td class="row info">
      <div class="image">
        <TipeProfileImage-stub url="http://placekitten.com/g/200/300"></TipeProfileImage-stub>
      </div>
      <div class="column">
        <p class="name-text">Garnett Koch</p>
        <p class="small-text email">Hank_Zemlak@hotmail.com</p>
      </div>
    </td>
    <td class="row">
      <TipeSelect-stub type="button" options="[object Object],[object Object],[object Object]" placeholder="" defaultSelected="[object Object]" name="" size="small" tabindex="0" dropdown="true"></TipeSelect-stub>
    </td>
    <td class="row">
      <p class="small-text"></p>
    </td>
    <td class="row">
      <p class="small-text">Offline</p>
    </td>
    <td class="row">
      <TipeButton-stub color="none" iconBefore="paper-airplane" iconAfter="" name="" size="icon" tabindex="0"></TipeButton-stub>
      <TipeButton-stub color="none" iconBefore="trash" iconAfter="" name="" size="icon" tabindex="0"></TipeButton-stub>
    </td>
  </tr>
  <Seperator-stub></Seperator-stub>
</div>
`;
